src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -43,6 +43,11 @@ import scala.runtime.Statics
   *  // 2
   *  // 3
   *  }}}
+  * 
+  *  A `LazyList`, like the one in the example above, may be infinite in length.
+  *  Aggregate methods, such as `count`, `sum`, `max` or `min` on such infinite length 
+  *  sequences will not terminate. Filtered infinite lazy lists are also effectively
+  *  infinite in length.
```

Comment:
It struck me yesterday that `infinite-sized` is redundant. WDYT? Similarly, I would use "infinite sequences", which also avoids wondering whether it should be hyphenated "infinite-length sequences". If you agree, I would remove "-sized" from the macro def while in the neighborhood. I'm not sure about "effectively infinite." It's infinite in just the same way. "Filtering an infinite lazy list results in an infinite collection for which aggregations will also not terminate." Or similar.

Reply:
Does an unevaluated LazyList contain any elements? How about infinitely many? Maybe an infinite list is defined by taking infinite time to produce its last element. Calling any `l.filter(_ => false)` infinite-sized is obviously false. Is calling it infinite any better? Maybe they should be called non-terminating collections named after the non-termination of evaluating any of its elements, and infinite collections are an example of that. Or does that become too tautological?

Reply:
I did think about what's implied by "infinite"; the doc says "create an infinite LazyList", so I went with it. I think the operation is non-terminating; the lazy collection just sits there. I agree one can't say filtering results in an infinite collection, so strike the second infinite. Also, the reporter's intuition is not obviously wrong (the filtered list is size 10 or maybe 9 and it should know that). So this bit of doc is very helpful.

Reply:
This is how "effectively infinite" was born. Not my proudest moment. But what *is* the overarching thing that `LazyList.continuously(1)` and `LazyList.continuously(1).filter(_ > 2)` can be called that results in the non-termination of aggregate methods? Or should we just accept that we may never find a good term describing that?

Reply:
I pushed an alternative, phrased around the termination of full iteration. Maybe it should still mention infiniteness as an example of non-termination of full iteration. Maybe that becomes so verbose the net positiveness becomes questionable. I'm not sure.

Reply:
I think I would go with "infinitely large collections", personally.

Reply:
while "non-terminating" is correct, that can also happen if your lazy element generation function doesn't return, which isn't really what the note is trying to convey (and I don't think it particularly adds any information that isn't already known)

Reply:
bikeshed: "unbounded"?

Reply:
"prometheus unbounded"?